src/mesh/amr_loadbalance.cpp

```diff
@@ -296,7 +298,9 @@ bool TryRecvSameToSame(int lid_recv, int send_rank, Variable<Real> *var, MeshBlo
       pmb->pmr->DereferenceCount() = counter_subview_h(0);
       var->dealloc_count = counter_subview_h(1);
     } else {
-      if (pmb->IsAllocated(var->label())) pmb->DeallocateSparse(var->label());
+      if (pmb->IsAllocated(var->label()) &&
+          !var->metadata().IsSet(Metadata::ForceAllocOnNewBlocks))
+        pmb->DeallocateSparse(var->label());
```

Comment:
Should there be an explicit allocation here if `Metadata::ForceAllocOnNewBlocks` is set?

Reply:
That's handled in `MeshBlockData::Initialize`